### PR TITLE
Slither action and avalanche-cli version

### PR DESCRIPTION
**Pull request by minghinmatthewlam**
## Why this should be merged
Fixes the slither action CI that is failing. New `crytic-compile` version is not handling the `target` set for `slither-action` correctly, and the slither action repo seems to be less managed/updated, so moving to slither command instead.

Also updates `avalanche-cli` version

## How this works
Install `slither-analyzer` which has the slither command through pip, and install foundry through `foundry-tools`. Will run slither on all src files and upload a sarif file that gets uploaded to produce alerts in the PRs.

## How this was tested
CI

## How is this documented

### CODE REVIEW

**Review by geoff-vball — APPROVED** [review]

File: contracts/foundry.toml [hunk 1]
```diff
@@ -2,5 +2,7 @@
 src = 'src'
 out = 'out'
 libs = ['lib']
+solc_version = '0.8.18'
+test = 'tests'
```
Comment (cam-schultz) [inline contracts/foundry.toml:6]: Are these two fields necessary for slither to work?

File: .github/workflows/slither.yml [hunk 2]
```diff
@@ -22,16 +22,27 @@ jobs:
...
+      - name: Run Slither
+        run: |
+          cd contracts
+          slither ./ --fail-none --sarif ./results.sarif
 
       - name: Upload SARIF file
         uses: github/codeql-action/upload-sarif@v2
         with:
-          sarif_file: ${{ steps.slither.outputs.sarif }}
+          sarif_file: contracts/results.sarif
```
Comment (cam-schultz) [inline .github/workflows/slither.yml:48]: I'm assuming that the above step creates a file `contracts/results.sarif`. We should add this to the `.gitignore` if it's not there already.

**Review by cam-schultz — APPROVED** [review]
LGTM. Just two small comments

File: contracts/foundry.toml (hunk 1 above)
Comment (minghinmatthewlam) [reply contracts/foundry.toml:6]: the first one was recommended to be set, but should be fine without. The second one is needed so slither doesn't run on test files and bombard us with alerts in the PRs.

File: .github/workflows/slither.yml (hunk 2 above)
Comment (minghinmatthewlam) [reply .github/workflows/slither.yml:48]: the `results.sarif` file should be within the github action runner

File: .github/workflows/slither.yml [hunk 3]
```diff
@@ -22,16 +22,27 @@ jobs:
...
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
         with:
-          target: "contracts/"
-          sarif: results.sarif
-          slither-config: "contracts/slither.config.json"
+          version: nightly
+          cache: false
+
+      - name: Run Slither
```
[the remark this answers is not in the document]
Comment (minghinmatthewlam) [reply .github/workflows/slither.yml:40]: slither uses crytic-compile, which uses foundry/forge for compiling.

File: .github/workflows/slither.yml [hunk 4]
```diff
@@ -22,16 +22,27 @@ jobs:
...
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install Slither
+        run: |
+          pip install slither-analyzer
+
+      - name: Install Foundry
```
[the remark this answers is not in the document]
Comment (minghinmatthewlam) [reply .github/workflows/slither.yml:34]: yup @geoff-vball also has a [PR](https://github.com/ava-labs/teleporter/pull/80) pinning the foundryup commit, so want to sync on this

File: .github/workflows/slither.yml (hunk 2 above)
Comment (geoff-vball) [reply .github/workflows/slither.yml:48]: Technically we should be able to run slither locally if we wanted to. In which case we would want it ignored.

File: .github/workflows/slither.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (geoff-vball) [reply .github/workflows/slither.yml:34]: Yeah I think this is a much simpler way of install foundry. It's supported by the foundry team. The only place we can't use this is inside the docker container for the integ tests, but that should be going away.

[the remark this answers is not in the document]
Comment (minghinmatthewlam) [reply .github/workflows/slither.yml:34]: yup think it's simpler to install, changed the other workflows to match as well.

**Review by geoff-vball — APPROVED** [review]
👍

**Review by michaelkaplan13 — APPROVED** [review]
LGTM. Thanks for working through this dependency issue.

**Pull request merged**